docs/sources/operators-guide/configure/setting-helm-ha-deduplication-consul/index.md

```diff
@@ -0,0 +1,161 @@
+    - name: Mimir
+      url: http://<mimir-endpoint>/prometheus
+      type: prometheus
+      editable: true
+```
+
+The above Helm yaml file will generate the following Prometheus configuration.
+
+```yaml
+global:
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  evaluation_interval: 1m
+  external_labels:
+    __replica__: prometheus-my-prometheus-kube-prometh-prometheus-x # x is the replica number
+    cluster: my-prometheus
+```
+
+You can execute the following command to install a release from the above chart.
+
+```bash
+helm upgrade --install -n <namespace> --create-namespace my-prometheus prometheus-community/kube-prometheus-stack --values values.yml
+```
+
+## Install Consul using Helm
+
+You can install Consul in Kubernetes using
+[Consul helm chart](https://github.com/hashicorp/consul-k8s/tree/main/charts/consul). Follow the documentation to get
+the chart and installing a Consul release. Put a note on the Consul endpoint, because you will need it for Mimir
+configuration.
+
+## Install/Upgrade Mimir config
+
+Follow [Getting Started with Helm]({{< relref "../../deploy-grafana-mimir/getting-started-helm-charts/_index.md" >}})
+documentation but add some configuration changes below.
+
+You can set Mimir HA deduplication configuration in global level or tenant level.
+
+### HA Deduplication Global
+
+To configure HA deduplication globally, add the following configuration below the `custom.yaml` file.
+
+```yaml
+# other configurations above
+mimir:
+  structuredConfig:
+    limits:
+      accept_ha_samples: true
+      # The following two configs must match with external_labels config in Prometheus
+      ha_cluster_label: cluster
+      ha_replica_label: __replica__
+    distributor:
+      ha_tracker:
+        enable_ha_tracker: true
+        kvstore:
+          store: consul
+          consul:
+            host: <consul_endpoint>
+```
+
+Make sure to install or upgrade the Mimir's helm release using the above configuration.
+
+```bash
+ helm -n mimir-test upgrade --install mimir grafana/mimir-distributed -f custom.yaml
+```
+
+### HA Deduplication per tenant
+
+TODO: Test this
+
+To configure HA deduplication per-tenant, add the following configuration below the `custom.yaml` file.
+
+```yaml
+runtimeConfig:
+  <tenant-id>:
+    accept_ha_samples: true
+    ha_cluster_label: cluster
+    ha_replica_label: __replica__
+```
+
+Make sure to install or upgrade the Mimir's helm release using the above configuration.
+
+## Verifying deduplication
+
+After Consul, Prometheus and Mimir running. Port forward Mimir distributor service.
+
+```bash
+kubectl port-forward service/mymimir-distributor 8080:8080
+```
+
+Open `http://localhost:8080/distributor/ha_tracker` in a browser. You should see the output similar like the following.
+If the table is empty, it means there is something wrong with the configuration.
+
+![HA Tracker Status](ha-tracker-status.png)
+
+Next, port forward Grafana that comes with kube-prometheus-stack.
```

Comment:
I think this (port fw grafana and data source) is again out of scope here, if we assume the user has Mimir set up already, they probably have Grafana as well.